pkg/scalers/aws_sqs_queue_scaler.go

```diff
@@ -21,37 +22,33 @@ import (
 
 const (
 	awsSqsQueueMetricName    = "ApproximateNumberOfMessages"
-	awsAccessKeyIDEnvVar     = "AWS_ACCESS_KEY_ID"
-	awsSecretAccessKeyEnvVar = "AWS_SECRET_ACCESS_KEY"
+	targetQueueLengthDefault = 5
 )
 
 type awsSqsQueueScaler struct {
-	metadata  *awsSqsQueueMetadata
-	sqsClient *sqs.SQS
+	metadata *awsSqsQueueMetadata
 }
 
 type awsSqsQueueMetadata struct {
-	targetQueueLength  int
-	queueURL           string
-	awsRegion          string
-	awsAccessKeyID     string
-	awsSecretAccessKey string
-	awsSessionToken    string
+	targetQueueLength int
+	queueURL          string
+	awsRegion         string
+	awsAuthorization  awsAuthorizationMetadata
 }
 
 var sqsQueueLog = logf.Log.WithName("aws_sqs_queue_scaler")
 
 // NewAwsSqsQueueScaler creates a new awsSqsQueueScaler
-func NewAwsSqsQueueScaler(resolvedEnv, metadata map[string]string) (Scaler, error) {
-	meta, err := parseAwsSqsQueueMetadata(metadata, resolvedEnv)
+func NewAwsSqsQueueScaler(resolvedEnv, metadata map[string]string, authParams map[string]string) (Scaler, error) {
+	meta, err := parseAwsSqsQueueMetadata(metadata, resolvedEnv, authParams)
 	if err != nil {
 		return nil, fmt.Errorf("Error parsing SQS queue metadata: %s", err)
 	}
 
 	sess := session.Must(session.NewSession())
 	if sess != nil {
 		s, err := session.NewSession(&aws.Config{
-			Credentials: credentials.NewStaticCredentials(meta.awsAccessKeyID, meta.awsSecretAccessKey, meta.awsSessionToken),
+			Credentials: credentials.NewStaticCredentials(meta.awsAuthorization.awsAccessKeyID, meta.awsAuthorization.awsSecretAccessKey, meta.awsAuthorization.awsSessionToken),
 			Region:      aws.String(meta.awsRegion),
 		})
```

Comment:
It looks like this whole section can be removed, right?

Reply:
It needs to stay. We have the credentials, now we need to use them. In the case of the role, we need to get temporary credentials and then use them.

Reply:
After re-reading. I see your point.